api/src/main/java/io/opencensus/stats/MeasureMap.java

```diff
@@ -51,6 +52,17 @@
    */
   public abstract MeasureMap put(MeasureLong measure, long value);
 
+  /**
+   * Associate the contextual information of an {@code Exemplar} to this {@link MeasureMap}.
+   *
+   * <p>If this method is called multiple times, only the last string map will be kept.
```

Comment:
Should we merge the attachments instead?  The keys allow users to add multiple types of exemplars, so I think it would be better to allow the exemplars to be added independently.  For example, a user could write helper methods:

```java
void addCurrentSpan(MeasureMap measures);
void addThrowable(MeasureMap measures, Throwable throwable);
```

Then `addCurrentSpan` could add keys such as "traceId" and "spanId", and `addThrowable` could add "errorMessage".

Reply:
Make sense to me, updated. And I think if a new value for an existing key is added to a `MeasureMap`, it should still overwrite the previous value (e.g putting a new TraceId should overwrite the previous one).

Reply:
Yes, I think it makes sense to overwrite the previous value.